contentgrid-spring-data-rest/src/main/java/com/contentgrid/spring/data/rest/webmvc/DefaultDomainTypeToHalFormsPayloadMetadataConverter.java

```diff
@@ -227,4 +244,34 @@ public Class<?> getType() {
         }
     }
 
+    @FunctionalInterface
+    static interface PropertyMetadataFactory {
+        PropertyMetadata build(Property property, Class<?> domainClass, String path);
+    }
+
+    private static String underscoreToCamelCase(String underscored) {
+        int index = underscored.indexOf('_');
+        if (index == -1) {
+            return underscored;
+        }
+
+        int from = 0;
+        StringBuilder camel = new StringBuilder();
+        while (index != -1) {
+            camel.append(underscored, from, index);
+            if (index + 1 < underscored.length()) {
+                camel.append((char) (underscored.charAt(index + 1) - 32));
+                from = index + 2;
```

Comment:
Can StringUtils.capitalize() be used in this place?

Reply:
No, because it's not a String. But there's `Character.toUpperCase()` that should be clearer than `- 32`.